app/src/main/java/com/example/drawingactivity/ColorUtils.kt

```diff
@@ -0,0 +1,75 @@
+package com.example.drawingactivity
+
+import android.graphics.Color
+import java.util.regex.Matcher
+import java.util.regex.Pattern
+
+object ColorUtils {
+
+    private val HEXADECIMAL_PATTERN: Pattern = Pattern.compile("\\p{XDigit}+")
+
+    /**
+     * A data class that contains color values.
+     * Contains: Hue, Saturation, Value, Red, Green, Blue, and Hex Value.
+     */
+    data class ColorState(
+        var hue: Int,
+        var saturation: Float,
+        var value: Float,
+        var red: Int,
+        var green: Int,
+        var blue: Int,
+        var hexCode: String
+    )
+
+    /**
+     * Calculates the RGB of a color using a HSV float array.
+     * Returns a ColorState containing all values.
+     */
+    fun calculateColorWithHsv(hsv: FloatArray): ColorState {
+        val currentColor: Int = Color.HSVToColor(hsv)
+        val hex = java.lang.String.format(
+            "#%02x%02x%02x",
+            Color.red(currentColor),
+            Color.green(currentColor),
+            Color.blue(currentColor)
+        )
+        return ColorState(
+            hue = hsv[0].toInt(),
+            saturation = hsv[1],
+            value = hsv[2],
+            red = Color.red(currentColor),
+            green = Color.green(currentColor),
+            blue = Color.blue(currentColor),
+            hexCode = hex
+        )
+    }
+
+    /**
+     * Calculates the HSV of a color using a RGB Int array.
+     * Returns a ColorState containing all values.
+     */
+    fun calculateColorWithRgb(rgb: IntArray): ColorState {
```

Comment:
This isn't super useful as the hue will never be "correct" here. I think all of our dealings should deal with HSV first. But I do see the need that you are letting them change the rgb fields, so maybe we don't need to be so stickler about it. But maybe we can look for ways to cut down this code. Essentially it's doing all the same things as the function above it, just with a different input. Very duplicate

Reply:
I feel like it is very similar because it doesn't need to provide anything other than view level control. I could possibly do some sort of singular function if I could just check what type of array was passed in the parameters, but I am unsure if that would be worthwhile in this case since a majority of the logic does run through the hsv calculations.